integreat_cms/cms/views/events/event_list_view.py

```diff
@@ -159,14 +160,36 @@ def get(self, request, *args, **kwargs):
                 ):
                     # Only non-recurring events
                     events = events.filter(recurrence_rule__isnull=True)
-                # Filter events by the search query
+                # Filter for upcoming or past events
+                if (
+                    len(event_filter_form.cleaned_data["time_range_events"])
+                    == len(time_range_events.CHOICES)
+                    or len(event_filter_form.cleaned_data["time_range_events"]) == 0
+                ):
+                    # Either all or no checkboxes are checked => skip filtering
+                    pass
+                elif (
+                    event_filter_form.cleaned_data["time_range_events"][0]
+                    == time_range_events.UPCOMING_EVENTS
+                ):
+                    # Only upcoming events
+                    events = events.filter(end_date__gte=date.today())
+                elif (
+                    event_filter_form.cleaned_data["time_range_events"][0]
+                    == time_range_events.PAST_EVENTS
+                ):
+                    # Only past events
+                    events = events.filter(end_date__lt=date.today())
+
+                # Filter events by the search querystart
```

Comment:
This looks like a typo
```suggestion
                # Filter events by the search query
```